mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -230,4 +264,87 @@ private int getClusteringColumnIndex(
       return -1;
     }
   }
+
+  private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
+
+    NamingStrategy namingStrategy = classElement.getAnnotation(NamingStrategy.class);
+    if (namingStrategy == null) {
+      return CqlNameGenerator.DEFAULT;
+    }
+
+    NamingConvention[] conventions = namingStrategy.convention();
+    TypeMirror[] customConverterClasses = readCustomConverterClasses(classElement);
+
+    if (conventions.length > 0 && customConverterClasses.length > 0) {
+      context
+          .getMessager()
+          .error(
+              classElement,
+              "Invalid annotation configuration: %s must have either a 'convention' "
+                  + "or 'customConverterClass' argument, but not both",
+              NamingStrategy.class.getSimpleName());
+      // Return a generator anyway, so that the processor doesn't crash downstream
+      return new CqlNameGenerator(conventions[0]);
+    } else if (conventions.length == 0 && customConverterClasses.length == 0) {
+      context
+          .getMessager()
+          .error(
+              classElement,
+              "Invalid annotation configuration: %s must have either a 'convention' "
+                  + "or 'customConverterClass' argument",
+              NamingStrategy.class.getSimpleName());
+      return CqlNameGenerator.DEFAULT;
+    } else if (conventions.length > 0) {
+      if (conventions.length > 1) {
+        context
+            .getMessager()
+            .warn(
+                classElement,
+                "Too many naming conventions: %s must have at most one 'convention' "
+                    + "argument (will use the first one: %s)",
+                NamingStrategy.class.getSimpleName(),
+                conventions[0]);
+      }
+      return new CqlNameGenerator(conventions[0]);
+    } else {
+      if (customConverterClasses.length > 1) {
+        context
+            .getMessager()
+            .warn(
+                classElement,
+                "Too many custom converters: %s must have at most one "
+                    + "'customConverterClass' argument (will use the first one: %s)",
+                NamingStrategy.class.getSimpleName(),
+                customConverterClasses[0]);
+      }
+      return new CqlNameGenerator(customConverterClasses[0]);
+    }
+  }
+
+  private TypeMirror[] readCustomConverterClasses(TypeElement classElement) {
+    // customConverterClass references a class that might not be compiled yet, so we can't read it
+    // directly, we need to go through mirrors.
+    AnnotationMirror annotationMirror = null;
+    for (AnnotationMirror candidate : classElement.getAnnotationMirrors()) {
```

Comment:
This is executed in a context of the specific class, right?
So we are getting only custom converters for a specific entity - not all converters for all classes?

Reply:
Yes, we are reading the converter class declared by the annotation for this entity (note that after [JAVA-2111](https://datastax-oss.atlassian.net/browse/JAVA-2111), the annotation might be inherited).